solr/solrj/src/java/org/apache/solr/client/solrj/request/AbstractUpdateRequest.java

```diff
@@ -21,10 +21,9 @@
 import org.apache.solr.common.params.ModifiableSolrParams;
 import org.apache.solr.common.params.UpdateParams;
 
-/** */
 public abstract class AbstractUpdateRequest extends CollectionRequiringSolrRequest<UpdateResponse>
     implements IsUpdateRequest {
-  protected ModifiableSolrParams params;
+  protected ModifiableSolrParams params = new ModifiableSolrParams(); // TODO make final
```

Comment:
Up for debate on eventually final.  That means we lose the setter, which is called in a ton of places, in lieu of calling `getParams().add(params)`.  I actually went down that path and shelved the work.  I was motivated for clarity on the getParams value never being potentially shared with something else.

Reply:
It's easy for me to enforce that setParams arg isn't null.  I'll do that.  setParams is called kinda a lot BTW, although I feel it's usually not needed.  I can do a follow-up draft PR to show you what it looks like if *nobody* calls it.